src/libos/src/fs/dev_fs/dev_sgx/mod.rs

```diff
@@ -164,9 +164,13 @@ impl DevSgx {
             #[cfg(feature = "dcap")]
             SGX_CMD_NUM_GET_DCAP_QUOTE_SIZE => {
                 let arg = nonbuiltin_cmd.arg_mut::<u32>()?;
-                let quote_size = SGX_DCAP_QUOTE_GENERATOR.get_quote_size();
-                unsafe {
-                    *arg = quote_size;
+                if SGX_DCAP_QUOTE_GENERATOR.is_ql_healthy() {
```

Comment:
The code could be:
if !SGX_DCAP_QUOTE_GENERATOR.is_ql_healthy() {
         return_errno!(EIO, "DCAP device not healthy");
}

let quote_size = SGX_DCAP_QUOTE_GENERATOR.get_quote_size();
unsafe {
    *arg = quote_size;
}